src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestUtilities.cs

```diff
@@ -116,10 +116,12 @@ internal static async Task<JsonWebKey> DecryptSymmetricPopKeyAsync(JsonWebTokenH
             var tokenDecryptionParameters = new TokenValidationParameters()
             {
                 TokenDecryptionKeys = decryptionKeys,
+#pragma warning disable CS0618 // TODO: If this is a valid use of RequireSignedTokens, perhaps this property should be left as not obsolete to not disrupt our consumers.
                 RequireSignedTokens = false,
```

Comment:
Hoping for feedback/explanation here for why we don't require tokens to be signed in this case? Does it have something to do with the SHR being signed as the cryptographic protection being relied on?

Reply:
PoP key can be represented as an encrypted JsonWebKey ("jwe" claim of a confirmation key). In this case it's only important to decrypt the JsonWebKey as we don't really care about validating claims and signature (it's not an access token).

https://datatracker.ietf.org/doc/html/rfc7800#section-3.3